src/app/shared/components/table-template/table-template.component.ts

```diff
@@ -124,4 +125,21 @@ export class TableTemplateComponent implements OnInit, OnChanges, OnDestroy {
 
     this.selectAllClicked.emit({ selectAll: this.selectAll});
   }
+
+  public selectAllInit() {
+    let itemCount = this.data.data.length;
```

Comment:
Should this assignment be move din the check at line 133?

Reply:
Yes it should, good catch.